pytorch_lightning/core/memory.py

```diff
@@ -182,14 +179,11 @@ class ModelSummary(object):
         0.530     Total estimated model params size (MB)
     """
 
-    MODE_TOP = "top"
-    MODE_FULL = "full"
-    MODE_DEFAULT = MODE_TOP
-    MODES = [MODE_FULL, MODE_TOP]
+    MODES = dict(top=0, full=-1)
 
-    def __init__(self, model, mode: str = MODE_DEFAULT):
+    def __init__(self, model, max_depth: int = -1):
```

Comment:
I believe this class can be used by itself

cc: @awaelchli 

So maybe the previous init argument should be deprecated too

Reply:
yes, the `ModelSummary` class can be used by itself and I've removed mentions to the `mode` argument, except for `    MODES = dict(top=0, full=-1)` which is still used outside to route the old mode argument and to maintain existing checks.

Reply:
@ManuelPalermo this would be a breaking change, so we can't outright change the parameter like this. it should go through a similar deprecation process

Reply:
I was following the suggestion from https://github.com/PyTorchLightning/pytorch-lightning/pull/8062#discussion_r655456826

Reply:
yes, by dropping it I meant remove it but of course deprecate first, then remove ^^ 
simply remapping mode to max_depth during deprecation phase :)

Reply:
ah, ok, sorry for the misunderstanding! I'll update the PR